src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllBytesAsync.cs

```diff
@@ -85,6 +85,21 @@ public Task ReadFileOver2GBAsync()
             return Assert.ThrowsAsync<IOException>(async () => await File.ReadAllBytesAsync(path));
         }
 
+        [Fact]
+        [OuterLoop]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/45954", TestPlatforms.Browser)]
+        public Task MaxArrayLengthAsync()
+        {
+            string path = GetTestFilePath();
+            using (FileStream fs = File.Create(path))
+            {
+                fs.SetLength(Array.MaxLength + 1L);
+            }
+
+            // File is too large for ReadAllBytes at once
+            return Assert.ThrowsAsync<IOException>(async () => await File.ReadAllBytesAsync(path));
```

Comment:
I suggest making the test async and awaiting this.  We don't care about the efficiency gains of avoiding the extra level of state machine here, and the extra diagnostics / debuggability of having this async frame in place might be helpful.